modules/unsupported/flatgeobuf/src/test/java/org/geotools/data/flatgeobuf/FlatGeobufDataStoreTest.java

```diff
@@ -881,6 +883,36 @@ public void readCountriesFids() throws IOException {
         }
     }
 
+    @Test(expected = NumberFormatException.class)
+    public void readCountriesFidsInvalid() throws IOException {
+        SimpleFeatureSource featureSource = getFeatureSource("countries");
+        SimpleFeatureType schema = featureSource.getSchema();
+        FilterFactory2 ff = CommonFactoryFinder.getFilterFactory2();
+        Filter filter = ff.id(ff.featureId("countries.noreallong"));
+        Query query = new Query(schema.getTypeName(), filter);
+        SimpleFeatureCollection featureCollection = featureSource.getFeatures(query);
+        try (SimpleFeatureIterator it = featureCollection.features()) {
+            while (it.hasNext()) {
+                it.next();
```

Comment:
If it reaches here, the test has actually failed, it matched a feature while the result should have been empty... or not? Something like this (may not pass QA, not sure if indentation is correct):


```suggestion
                SimpleFeature f = it.next();
                fail("Unexpected feature match" + f);
```

Reply:
And no, it should not throw a number format, you should be getting nothing. is.hasNext() returns false and the loop exists.

Reply:
Ah.. yes, I should filter them out.